openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -157,347 +50,120 @@ def override_for_request(self, switch_name, active=True):
+        # not all contained within the __init__ module.
+        from openedx.core.djangoapps.waffle_utils.testutils import override_waffle_flag
 
-        """
-        if name not in _WAFFLE_FLAG_CUSTOM_ATTRIBUTE_SET:
-            return
+        with override_waffle_flag(self, active):
+            yield
 
-        flag_attribute_data = self._get_request_cache().setdefault('flag_attribute', {})
-        is_value_change = False
-        if name not in flag_attribute_data:
-            flag_attribute_data[name] = str(value)
-            is_value_change = True
-        else:
-            if flag_attribute_data[name] != str(value):
-                flag_attribute_data[name] = 'Both'
-                is_value_change = True
 
-        if is_value_change:
-            attribute_name = 'flag_{}'.format(name)
-            set_custom_attribute(attribute_name, flag_attribute_data[name])
+class CourseWaffleFlag(WaffleFlag):
+    """
+    Represents a single waffle flag that can be forced on/off for a course. This class should be used instead of
+    WaffleFlag when in the context of a course.
 
+    Uses a cached waffle namespace.
 
-def _get_waffle_flag_custom_attributes_set():
-    """
-    Returns a set based on the Django setting WAFFLE_FLAG_CUSTOM_ATTRIBUTES (list).
-    """
-    waffle_flag_custom_attributes = getattr(settings, _WAFFLE_FLAG_CUSTOM_ATTRIBUTES, None)
-    waffle_flag_custom_attributes = waffle_flag_custom_attributes if waffle_flag_custom_attributes else []
-    return set(waffle_flag_custom_attributes)
+    Usage:
 
-# .. toggle_name: WAFFLE_FLAG_CUSTOM_ATTRIBUTES
-# .. toggle_implementation: DjangoSetting
-# .. toggle_default: False
-# .. toggle_description: A list of waffle flags to track with custom attributes having
-#   values of (True, False, or Both).
-# .. toggle_use_cases: opt_in
-# .. toggle_creation_date: 2020-06-17
-# .. toggle_warnings: Intent is for temporary research (1 day - several weeks) of a flag's usage.
-_WAFFLE_FLAG_CUSTOM_ATTRIBUTES = 'WAFFLE_FLAG_CUSTOM_ATTRIBUTES'
+       WAFFLE_FLAG_NAMESPACE = WaffleFlagNamespace(name='my_namespace')
+       SOME_COURSE_FLAG = CourseWaffleFlag(WAFFLE_FLAG_NAMESPACE, 'some_course_feature', __name__)
 
-# set of waffle flags that should be instrumented with custom attributes
-_WAFFLE_FLAG_CUSTOM_ATTRIBUTE_SET = _get_waffle_flag_custom_attributes_set()
+    And then we can check this flag in code with::
 
+        SOME_COURSE_FLAG.is_enabled(course_key)
 
-class WaffleFlag(object):
-    """
-    Represents a single waffle flag, using a cached waffle namespace.
+    The Django Admin "waffle_utils" section can be used to configure a course override for this same flag (e.g.
+    my_namespace.some_course_feature).
     """
-    # use a WeakSet so these instances can be garbage collected if need be
-    _class_instances = WeakSet()
 
-    def __init__(self, waffle_namespace, flag_name, module_name=None):
+    def _get_course_override_value(self, course_key):
```

Comment:
Oops. I left a comment in a later PR: https://github.com/edx/edx-platform/pull/25417#discussion_r513890482